Lib/sqlite3/test/regression.py

```diff
@@ -376,6 +376,37 @@ def CheckCommitCursorReset(self):
                 counter += 1
         self.assertEqual(counter, 3, "should have returned exactly three rows")
 
+    def CheckUninitializedIsolationLevel(self):
+        """
+        Previously trying to get the isolation level of an uninitialized Connection
+        caused a segfault, now it should just return None.
+        """
+        conn = sqlite.Connection.__new__(sqlite.Connection)
+        self.assertEqual(conn.isolation_level, None)
+
+
+    def CheckCursorInvalidIsolationLevel(self):
+        """
+        When trying to call conn.corsor() when conn is a Connection object that
+        was not initialized properly, it caused a segfault. Now it should raise
+        a ProgrammingError.
+        """
+        try:
+            conn = sqlite.Connection.__new__(sqlite.Connection)
+            conn.__init__('', isolation_level='invalid isolation level')
+        except ValueError:
```

Comment:
It would be better to wrap the `__init__` call with `with self.assertRaises(ValueError):` here.